mlflow/langchain/__init__.py

```diff
@@ -314,19 +328,19 @@ def load_retriever(persist_directory):
             **model_data_kwargs,
         }
     else:
-        # TODO: use model_config instead
         # If the model is a string, we expect the code_path which is ideally config.yml
         # would be used in the model. We set the code_path here so it can be set
         # globally when the model is loaded with the local path. So the consumer
         # can use that path instead of the config.yml path when the model is loaded
+        # TODO: what if model_config is not a string / file path?
         flavor_conf = (
-            {_CODE_CONFIG: code_paths[0], _CODE_PATH: lc_model}
-            if code_paths and len(code_paths) >= 1
-            else {_CODE_CONFIG: None, _CODE_PATH: lc_model}
+            {_MODEL_CODE_CONFIG: model_config_path, _MODEL_CODE_PATH: lc_model}
+            if model_config_path
```

Comment:
model_config_path uses model_config primarily, falls back to code_paths[0] for backwards compatibility. will deprecate usage of code_paths[0] in the future